### ISSUE TITLE
Storage Access Headers

### ISSUE THREAD

**Issue by cfredric** [WebKit/standards-positions#412]
### WebKittens

@annevk @johnwilander

### Title of the proposal

Storage Access Headers

### URL to the spec

https://privacycg.github.io/storage-access-headers/

### URL to the spec's repository

https://github.com/privacycg/storage-access-headers

### TAG Design Review URL

https://github.com/w3ctag/design-reviews/issues/982

### Mozilla standards-positions issue URL

https://github.com/mozilla/standards-positions/issues/1084

### Description

The Storage Access Headers proposal creates new HTTP request and response headers to enable authenticated embeds to access third-party cookies, even without an iframe, via existing storage-access permission grants.

**Comment by othermaciej** [comment]
The “spec” here is just a skeleton pointing to the explainer: https://github.com/privacycg/storage-access-headers

**Comment by johannhof** [comment]
Hi Maciej, thanks for taking a look! Yes, it's a placeholder. I think @cfredric has been hacking away on an initial spec in his personal fork of the repo that we'll try to merge some time this week.

**Comment by cfredric** [comment]
Yes indeed. I've been working on a draft spec in https://github.com/privacycg/storage-access-headers/pull/20.

**Comment by RByers** [comment]
Note that that this spec is now [adopted by the privacy CG](https://privacycg.github.io/storage-access-headers/) and Chromium has just [approved](https://groups.google.com/a/chromium.org/g/blink-dev/c/gERgwZfN_-E/m/QIJbT1zUCgAJ) shipping it.

**Comment by annevk** [comment]
It seems that generally Storage Access Headers preserves the invariants colleagues and I care about and as such is a reasonable extension of the Storage Access feature.

Unfortunately it's hard to fully evaluate however as there's still a number of missing pieces to the standardization story of Storage Access itself. One thing I did spot is that worker integration appears to be completely missing: https://github.com/privacycg/storage-access-headers/issues/26.

**Comment by johannhof** [comment]
Thanks Anne, that makes sense! I think we agree that there are some things we need to better figure out around Storage Access, particularly workers. I do expect that to come out of the collaboration we already have on both SAA and Cookie Layering :)

**Comment by annevk** [comment]
Modulo resolving worker integration colleagues and I are supportive of this effort and as such I'd like to suggest we mark it as "position: support" two weeks from now. If anyone else has any comments please comment here or reach out to me directly. Thanks!
